src/taco-utils/npmHelper.ts

```diff
@@ -0,0 +1,50 @@
+/**
+﻿ *******************************************************
+﻿ *                                                     *
+﻿ *   Copyright (C) Microsoft. All rights reserved.     *
+﻿ *                                                     *
+﻿ *******************************************************
+﻿ */
+
+/// <reference path="../typings/node.d.ts" />
+/// <reference path="../typings/npm.d.ts" />
+/// <reference path="../typings/q.d.ts" />
+
+"use strict";
+
+import npm = require("npm");
+import path = require("path");
+import installLogLevel = require("./installLogLevel");
+import Q = require("q");
+
+import InstallLogLevel = installLogLevel.InstallLogLevel;
+
+module TacoUtility {
+	export class NpmHelper {
+		private static runNpmCommand(npmCommand: string, path: string, args: string[], optionalArgs: any, logLevel?: InstallLogLevel): Q.Promise<any> {
+			return Q.nfcall(npm.load, {}).then(function() {
```

Comment:
So in the end we decided against having a single npm.load and went with a load-per-operation? Is that because load is a no-op for npm after the first time?